### PR TITLE
Add blog post for crossplane

### PR DESCRIPTION
**Pull request by echarles**
This add a blog post for crossplane.

Any feedback on this draft blog post @bassam @jbw976 @hasheddan @stevendborrelli @grantgumina @muvaf ?

### CODE REVIEW

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+date: 2021-05-16
+author: Eric Charles
+author_title: Datalayer Founder
+author_url: https://github.com/echarles
+author_image_url: https://en.gravatar.com/userimage/5834995/1e62fb4fa4f0986227cc8e5a0dbbf3cf.png
+tags: [datalayer, crossplane, upbound, kubernetes, control-plane, xrd, crd, helm]
+hide_table_of_contents: false
+---
+
+[Crossplane](https://crossplane.io) is an open source Kubernetes add-on that enables platform teams to assemble infrastructure from multiple vendors, and expose higher level self-service APIs for application teams to consume, without having to write any code. It allows you to compose cloud infrastructure and services based on `XRD` (cross resource definitions) that extends the existing Kubernete CRD (Custom Resource Definition). To achieve this awesome goal, you have to use various repositories that reside in the GitHub [crossplane](https://github.com/crossplane), [crossplane-contrib](https://github.com/crossplane-contrib) and [upbound](https://github.com/upbound) organisations. As adaptor of that new technology, you can rely the officiel [documentation](https://crossplane.github.io/docs/v1.2) where a lot of details are gathered.
```
Comment (jbw976) [inline blog/2021-05-16-crossplane-by-example.mdx:13]: officiel --> official

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+On top of the managed infrastructure, you can create custom [compositions](https://crossplane.io/docs/v1.2/concepts/composition.html) that ships `platform` composed of managed infrastructures and services. A platform could be for example the sum of a Kubernetes cluser, a Database, a Bucket and Helm charts deployed on the Kubernetes cluster, all working together, meaning that the application would connect to the Database and the Bucket. There are already a few of those [reference platforms](https://github.com/upbound?q=ref) available.
+
+To navigate the growing CRD/XRD definitions, you can use a [documentation website for CRD/XRD](https://doc.crds.dev). You give the repository hosting the RDC/XRD and the website will introspect and generate for you a nice documentation page, see for example [the generated documentation for the Google cloud provider](https://doc.crds.dev/github.com/crossplane/provider-gcp).
```
Comment (jbw976) [inline blog/2021-05-16-crossplane-by-example.mdx:57]: RDC --> CRD?

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Back in 2018, we have developed the very first interation of the [Datalayer platform on Kubernetes](https://blog.datalayer.io/2018/01/16/datalayer-on-k8s) with home-made forged tools like [kuber-go](https://github.com/datalayer-attic/kuber-go), now in the attic, to deploy and manage Kubernetes on various clouds.
+
+We have continued looking at tools like:
+
+- [Terraform](https://www.terraform.io) which is in the IaC (Infrastructure as Code) vein, with proprietary language.
+- [Pulumi](https://www.pulumi.com) is also IaC with multi-language support but needing an account on the pulumi systems.
+- [Cluster API](https://cluster-api.sigs.k8s.io) which is limited to the Kubernetes cluster creation.
+
+Instead of those standard approach, we have been thrilled by the pure declarative way, a-la-kubernetes, that Crossplane provides to operate the cloud infrastructure and services you need. So instead of building our own *cloud-chain toolkit*, we have decided to embrass the GitOps principles with YAML declarations that allow to create all kind of infrastructure and service resources. The reconciliation loops implemented by Kubernetes Crossplane operators ensure that any drift is fixed.
```
Comment (jbw976) [inline blog/2021-05-16-crossplane-by-example.mdx:45]: embrass --> embrace?

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+- Finally, the open-source repositories are backed by a commercial company, [Upbound](https://upbound.io), which also operates [a cloud](https://cloud.upbound.io) to host your custom configurations.
+
+## ✈️ A Plane for Datalayer
+
+At Datalayer, we are pivoting our cloud solutions to leverage the Crossplane goodies with 2 tracks:
+
+1. Developing `clouder`, a software component to better integrate Crossplane with [Flux](https://fluxcd.io), a CI/CD solution that makes your GitOps flow a breeze. Clouder will deliver easier bootstraping as a WEB user interface for operators to manage and monitor the platform.
+
+2. Building an opiniated configuration platform `jupyter-gcp-plane` for [Jupyter notebooks](https://jupyter.org), reusing the existing [JupyterHub Helm chart](https://zero-to-jupyterhub.readthedocs.io).
```
Comment (jbw976) [inline blog/2021-05-16-crossplane-by-example.mdx:79]: love this idea for a reference platform for Jupyter notebooks!

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+- Some customers claim to run it in production environment.
+
+- Finally, the open-source repositories are backed by a commercial company, [Upbound](https://upbound.io), which also operates [a cloud](https://cloud.upbound.io) to host your custom configurations.
+
+## ✈️ A Plane for Datalayer
+
+At Datalayer, we are pivoting our cloud solutions to leverage the Crossplane goodies with 2 tracks:
+
+1. Developing `clouder`, a software component to better integrate Crossplane with [Flux](https://fluxcd.io), a CI/CD solution that makes your GitOps flow a breeze. Clouder will deliver easier bootstraping as a WEB user interface for operators to manage and monitor the platform.
```
Comment (jbw976) [inline blog/2021-05-16-crossplane-by-example.mdx:77]: definitely excited to see progress on this tool too :)

**Review by jbw976 — COMMENTED** [review]
this is great @echarles! Really excited to see you embrace the crossplane project and bringing some enthusiasm and evangelism to the community 💪

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 1 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx:13]: Fixed. Thx.

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 2 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx:57]: Fixed. Thx.

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 4 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx:79]: Yeah, having more jupyter cloud-native (with database and bucket storage) can be useful.

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 3 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx:45]: Fixed. Thx.

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 5 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx:77]: It is still under design based on some experiments... BTW I see you have surfaced the flux/crossplane interaction in https://www.youtube.com/watch?v=38rSsVYhK-w&list=PLj6h78yzYM2OFWBatWHbWgyoLNmCyAqJ0&index=13 - I would love having an automated way to upload the key in the github repo, a bit like flux bootstrap / flux create source github  does it for you. 

Bootstraping and UI are the 2 identified aspects that desserve enhancement. WDYT?

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+Instead of those standard approach, we have been thrilled by the pure declarative way, a-la-kubernetes, that Crossplane provides to operate the cloud infrastructure and services you need. So instead of building our own *cloud-chain toolkit*, we have decided to embrace the GitOps principles with YAML declarations that allow to create all kind of infrastructure and service resources. The reconciliation loops implemented by Kubernetes Crossplane operators ensure that any drift is fixed.
+
+If your are *allergic* to YAML, which we can understand, you may pass your turn. If you really can not live without Terraform, you will be interested in some [work in progress](https://github.com/crossplane-contrib?q=terraform) to better support the Terraform ecosystem.
+
+## A rich ecosystem
+
+Cloud providers are available to create managed infrastructure, like Kubernetes clusters, databases, IAM roles, service accounts... on the major clouds like [Google cloud](https://github.com/crossplane/provider-gcp), [Amazon AWS](https://github.com/crossplane/provider-aws), [Microsoft Azure](https://github.com/crossplane/provider-asure)...
```
Comment (muvaf) [inline blog/2021-05-16-crossplane-by-example.mdx:51]:
```suggestion
Cloud providers are available to create managed infrastructure, like Kubernetes clusters, databases, IAM roles, service accounts... on the major clouds like [Google cloud](https://github.com/crossplane/provider-gcp), [Amazon AWS](https://github.com/crossplane/provider-aws), [Microsoft Azure](https://github.com/crossplane/provider-azure)...
```

File: blog/2021-05-16-crossplane-by-example.mdx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+To navigate the growing CRD/XRD definitions, you can use a [documentation website for CRD/XRD](https://doc.crds.dev). You give the repository hosting the CRD/XRD and the website will introspect and generate for you a nice documentation page, see for example [the generated documentation for the Google cloud provider](https://doc.crds.dev/github.com/crossplane/provider-gcp).
+
+## Future-proof
+
+If you are still reluctant to invest in that burgeoning technology, here are a few arguments.
+
+- Crossplane is already a [CNCF](https://www.cncf.io) sandbox project, being now proposed to [migrate to incubation status](https://github.com/cncf/toc/pull/620).
+
+- The lastest Kubecon Europe 2021 has shown [a lot of participants and traction](https://blog.upbound.io/crossplane-community-day-europe-2021-recap/).
```
Comment (muvaf) [inline blog/2021-05-16-crossplane-by-example.mdx:65]:
```suggestion
- The latest Kubecon Europe 2021 has shown [a lot of participants and traction](https://blog.upbound.io/crossplane-community-day-europe-2021-recap/).
```

**Review by muvaf — COMMENTED** [review]
Just some small typos. Thank you for all the links, it's great to see all in one place!

File: blog/2021-05-16-crossplane-by-example.mdx (hunk 5 above)
Comment (echarles) [reply blog/2021-05-16-crossplane-by-example.mdx]: The `crossfllux` repo from @hasheddan looks like a great step forward. I was thinking to something like `clouder --bootstrap` that would setup the flux github repos, the control cluster and a UI for infra administrator. I will follow the TBS for more news and publish stuff when I have something.

**Pull request merged**